### DISCUSSION TITLE
Use the image offers by microsoft or the official image?

### DISCUSSION THREAD

**Discussion by Standard-IO** [devcontainers/spec#183] [Q&A] [answered]
Reading these [docs](https://code.visualstudio.com/docs/devcontainers/containers#_create-a-devcontainerjson-file) says that:

> You can use any image, Dockerfile, or set of Docker Compose files as a starting point.

I have a Dockerfile that uses ubuntu official image. But I'am seeing that exist a ubuntu dev image offers by Microsoft as starting point (` mcr.microsoft.com/devcontainers/base:jammy` ). Does exist any adventage to use that image over the official one?

**Comment by samruddhikhandale** [comment, marked as answer]
@Standard-IO  👋 

Using the `mcr.microsoft.com/devcontainers/*` images has few advantages -

- It installs/upgrades the most common utils (tools/packages)
- Adds default bash/zsh themes
- Configures user
- Adds code IDE setup
- We try to provide a complete dev environment with the images/templates
- ❗ We are on top of detecting & adding patches to fix security vulnerabilities in the images ❗

**Reply by Standard-IO** [reply]
Thanks!, seems very well to me and  the last point is a strong one.
